traits/observers/_interfaces.py

```diff
@@ -0,0 +1,29 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+
+class IObserver:
```

Comment:
I think I was aiming at the second option there. Will use `abc.ABC`.

Reply:
Sounds good. I'm happy with the "this is actually an interface, but we'll also use it as a base class because it's convenient to do so" approach; that's the sort of (ab)use that's directly encouraged by the way that Python does it.